contracts/src/rollup/RollupCore.sol

```diff
@@ -402,51 +402,36 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
         bytes32 sequencerBatchAcc;
         {
             // This new assertion consumes the messages from prevInboxPosition to afterInboxPosition
-            uint64 afterInboxPosition = assertion.afterState.globalState.getInboxPosition();
-            uint64 prevInboxPosition = assertion.beforeState.globalState.getInboxPosition();
-
-            // there are 2 kinds of assertions that can be made.
-            // ERRORED: this is an unexpected state and an upgrade is required to continue from this state
-            // FINISHED: A normal assertion is made when all messages up to the prev.nextInboxPosition have been processed
-            //           This assertion MUST process all of those messages, therefore we expect it to have inbox position equal
-            //           to the prev.nextInboxPosition, and positionInMessage == 0.
+            GlobalState calldata afterGS = assertion.afterState.globalState;
+            GlobalState calldata beforeGS = assertion.beforeState.globalState;
+
+            // there are 3 kinds of assertions that can be made. Assertions must be made when they fill the maximum number
+            // of blocks, or when they process all messages up to prev.nextInboxPosition. When they fill the max
+            // blocks, but dont manage to process all messages, we call this an "overflow" assertion.
+            // 1. ERRORED assertion
+            //    The machine finished in an ERRORED state. This can happen with processing any
+            //    messages, or moving the position in the message.
+            // 2. FINISHED assertion that did not overflow
+            //    The machine finished as normal, and fully processed all the messages up to prev.nextInboxPosition.
+            //    In this case the inbox position must equal prev.nextInboxPosition and position in message must be 0
+            // 3. FINISHED assertion that did overflow
+            //    The machine finished as normal, but didn't process all messages in the inbox.
+            //    The inbox position can be anything (within valid range) except the prev.nextInboxPosition,
```

Comment:
Maybe we can make the range more specific. It has to be `prev.inboxPosition < N < prev.nextInboxPosition`, right?

Reply:
Good catch, this was already enforced in the code but the comment was incorrect. This comment was copied from before when we removed the overflow assertions.